src/App/Pages/Accounts/LoginSsoPageViewModel.cs

```diff
@@ -0,0 +1,177 @@
+            _platformUtilsService = ServiceContainer.Resolve<IPlatformUtilsService>("platformUtilsService");
+            _stateService = ServiceContainer.Resolve<IStateService>("stateService");
+
+            PageTitle = AppResources.Bitwarden;
+            LogInCommand = new Command(async () => await LogInAsync());
+        }
+
+        public string OrgIdentifier
+        {
+            get => _orgIdentifier;
+            set => SetProperty(ref _orgIdentifier, value);
+        }
+
+        public Command LogInCommand { get; }
+        public bool RememberOrgIdentifier { get; set; }
+        public Action StartTwoFactorAction { get; set; }
+        public Action StartSetPasswordAction { get; set; }
+        public Action LoggedInSsoAction { get; set; }
+        public Action CloseAction { get; set; }
+
+        public async Task InitAsync()
+        {
+            if (string.IsNullOrWhiteSpace(OrgIdentifier))
+            {
+                OrgIdentifier = await _storageService.GetAsync<string>(Keys_RememberedOrgIdentifier);
+            }
+            var rememberOrgIdentifier = await _storageService.GetAsync<bool?>(Keys_RememberOrgIdentifier);
+            RememberOrgIdentifier = rememberOrgIdentifier.GetValueOrDefault(true);
+        }
+
+        public async Task LogInAsync()
+        {
+            if (Connectivity.NetworkAccess == NetworkAccess.None)
+            {
+                await _platformUtilsService.ShowDialogAsync(AppResources.InternetConnectionRequiredMessage,
+                    AppResources.InternetConnectionRequiredTitle);
+                return;
+            }
+            if (string.IsNullOrWhiteSpace(OrgIdentifier))
+            {
+                await _platformUtilsService.ShowDialogAsync(
+                    string.Format(AppResources.ValidationFieldRequired, AppResources.OrgIdentifier),
+                    AppResources.AnErrorHasOccurred,
+                    AppResources.Ok);
+                return;
+            }
+
+            await _deviceActionService.ShowLoadingAsync(AppResources.LoggingIn);
+
+            var passwordOptions = new PasswordGenerationOptions(true);
+            passwordOptions.Length = 64;
+
+            var codeVerifier = await _passwordGenerationService.GeneratePasswordAsync(passwordOptions);
+            var codeVerifierHash = await _cryptoFunctionService.HashAsync(codeVerifier, CryptoHashAlgorithm.Sha256);
+            var codeChallenge = CoreHelpers.Base64UrlEncode(codeVerifierHash);
+            await _storageService.SaveAsync(Constants.SsoCodeVerifierKey, codeVerifier);
+
+            var state = await _passwordGenerationService.GeneratePasswordAsync(passwordOptions);
+            await _storageService.SaveAsync(Constants.SsoStateKey, state);
+
+            var redirectUri = "bitwarden://sso-callback";
+
+            var url = _apiService.IdentityBaseUrl + "/connect/authorize?" +
+                      "client_id=" + _platformUtilsService.IdentityClientId + "&" +
```

Comment:
I forgot about _platformUtilsService.IdentityClientId. I dont think we use that on other clients.

Reply:
Should I be using something else?

Reply:
No, this is fine.